aeppl/opt.py

```diff
@@ -185,44 +189,506 @@ def rng_fn(self, rng, *args, **kwargs):
         raise NotImplementedError()
 
 
-@local_optimizer([Scan])
-def scan_replace(fgraph, node):
+FieldInfo = namedtuple(
+    "FieldInfo", ("name", "agg_name", "index", "inner_index", "agg_index")
+)
 
-    node_outs = node.op.outputs
 
-    if not node_outs:
-        return
+def default_filter_scanargs(x):
+    return x.startswith("inner_") or x.startswith("outer_")
+
+
+class ScanArgs(scan_args):
```

Comment:
This can go in a `utils` module, or, if a lot more code is needed for the `Scan` work, its own module entirely.

Also, we should replace Aesara's current `scan_args` with this `ScanArgs`.

Reply:
I'll make a quick PR for `ScanArgs` in Aesara.

Reply:
Did it in https://github.com/aesara-devs/aesara/pull/510